pkg/operation/seed/seed.go

```diff
@@ -761,6 +778,24 @@ func BootstrapCluster(ctx context.Context, k8sGardenClient, k8sSeedClient kubern
 		return err
 	}
 
+	// managed nginx-ingress handling
+	if err := handleIngressDNSEntry(ctx, k8sSeedClient, chartApplier, seed); err != nil {
+		return err
+	}
+
+	var ingressClass string
+	if managedIngress(seed) {
+		ingressClass = v1beta1constants.SeedNginxIngressClass
+	} else {
+		ingressClass = v1beta1constants.ShootNginxIngressClass
+		if err := deleteIngressController(ctx, k8sSeedClient.Client()); err != nil {
+			return err
+		}
+	}
+	if err := migrateIngressClassForShootIngresses(ctx, k8sGardenClient.Client(), k8sSeedClient.Client(), seed, ingressClass); err != nil {
```

Comment:
Good point, but out of scope of this PR. This can be handled separately (you can open an issue if you want) but is not related to the managed ingress feature that @BeckerMax contibutes here.